openeo/rest/udp.py

```diff
@@ -31,20 +34,38 @@ def build_process_dict(
     :param description: detailed description
     :param parameters: list of process parameters (which have name, schema, default value, ...)
     :param returns: description and schema of what process returns
+    :param categories: list of categories
+    :param examples: list of examples, may be used for unit tests
+    :param links: list of links related to the process
     :return: dictionary in openEO "process graph with metadata" format
     """
     process = dict_no_none(
         process_graph=as_flat_graph(process_graph),
         id=process_id,
         summary=summary,
         description=description,
-        returns=returns
+        returns=returns,
     )
     if parameters is not None:
         process["parameters"] = [
             (p if isinstance(p, Parameter) else Parameter(**p)).to_dict()
             for p in parameters
         ]
+    if categories is not None:
+        process["categories"] = [
+            (c if isinstance(c, str) else str(c))
+            for c in categories
+        ]
+    if examples is not None:
+        process["examples"] = [
+            (e if isinstance(e, dict) else e.to_dict())
+            for e in examples
+        ]
+    if links is not None:
+        process["links"] = [
+            (l if isinstance(l, dict) else l.to_dict())
+            for l in links
+        ]
```

Comment:
Why is all this `... if isinstance ... to_dict ... for x in ...` necessary? 
The type hint already documents that e.g. `categories` is a list of strings, so why the need to for extra conversion?

Reply:
note the `to_dict` stuff is necessary for "parameters", because these can be `Parameter` objects which are not JSON-serializable. But for categories, examples, links there is no need for explicit conversion here because lists/dicts are JSON-serializable out of the box

Reply:
Alright, I thought it had something to do with the `Lists`. I removed it now.